File: fedu/src/main/java/com/fedu/fedu/entity/Subject.java
package com.fedu.fedu.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "subjects")
public class Subject extends AbstractEntity<Long> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "subject_id")
    private Long subjectId;

    @Column(name = "subject_code", unique = true, nullable = false)
    private String subjectCode;

    @Column(name = "subject_name", nullable = false)
    private String subjectName;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    /** Số chặng (stage) của lộ trình. Giới hạn stage_order của node thuộc môn. */
    @Column(name = "learningpath_length")
    private Integer learningpathLength;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private UserAccount createdBy;

    @Builder.Default
    @Column(name = "is_deleted")
    private Boolean isDeleted = false;

    @Builder.Default
    @Column(name = "status", nullable = false, length = 50)
    private String status = "draft";
}

File: SWP391_G1_FEdu_Fe/src/types/subject.ts


export interface Subject {
  subjectId: number;
  subjectCode: string;
  subjectName: string;
  description?: string;
  isDeleted?: boolean;
  createdBy?: {
    userId: number;
    firstName: string;
    lastName: string;
    email: string;
  } | null;
  createdAt?: string;
  updatedAt?: string;
  status?: string;
  learningpathLength?: number;
}

export interface SubjectRequest {
  subjectCode: string;
  subjectName: string;
  description?: string;
  status?: string;
  learningpathLength?: number;
}
